### PR TITLE
docs(book): tracking state in ExExes

### PR DESCRIPTION
**Pull request by shekhirin**

### CODE REVIEW

File: book/developers/exex/tracking-state.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        let handle = builder
+            .node(EthereumNode::default())
+            .install_exex("my-exex", |ctx| async move { Ok(MyExEx::new(ctx)) })
+            .launch()
+            .await?;
+
+        handle.wait_for_node_exit().await
+    })
+}
+```
```
Comment (shekhirin) [inline book/developers/exex/tracking-state.md:176]: I don't like that we basically repeat the whole snippet twice. How can we do it better?

File: book/developers/exex/tracking-state.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+use reth_tracing::tracing::info;
+
+struct MyExEx<Node: FullNodeComponents> {
+    ctx: ExExContext<Node>,
+}
+
+impl<Node: FullNodeComponents> Future for MyExEx<Node> {
+    type Output = eyre::Result<()>;
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
```
Comment (gakonst) [inline book/developers/exex/tracking-state.md:40]: We should also explain how you'd call out to other functions?

Prob we'd save most of the logic in async functions on the exex that the future loop calls?

File: book/developers/exex/tracking-state.md (hunk 1 above)
Comment (gakonst) [reply book/developers/exex/tracking-state.md:176]: https://rust-lang.github.io/mdBook/format/mdbook.html#including-files

**Review by gakonst — APPROVED** [review]

**Review by onbjerg — APPROVED** [review]

**Review by rkrasiuk — APPROVED** [review]

File: book/developers/exex/tracking-state.md (hunk 1 above)
Comment (shekhirin) [reply book/developers/exex/tracking-state.md:176]: we don't fully repeat the snippet but rather change some parts of it. I meant how we can do better at showing what changed in the code without repeating it all over again, but we can solve it later.

**Pull request merged**